service/evidence/evidence_store.go

```diff
@@ -135,22 +206,43 @@ func (svc *Service) StoreEvidence(ctx context.Context, req *evidence.StoreEviden
 		return nil, service.ErrPermissionDenied
 	}
 
+	// Store evidence
 	err = svc.storage.Create(req.Evidence)
 	if err != nil && errors.Is(err, persistence.ErrUniqueConstraintFailed) {
-		return nil, status.Error(codes.AlreadyExists, "entry already exists")
+		return nil, status.Error(codes.AlreadyExists, persistence.ErrEntryAlreadyExists.Error())
 	} else if err != nil {
-		return nil, status.Errorf(codes.Internal, "database error: %v", err)
+		return nil, status.Errorf(codes.Internal, "%v: %v", persistence.ErrDatabase, err)
 	}
 
 	go svc.informHooks(ctx, req.Evidence, nil)
 
+	// Send evidence to the channel for processing (fire and forget)
```

Comment:
The term "fire and forget", I would rather expect in the discovery because it fires (stores the evidence) and forget about it. Here, in the Evidence Store, we store the evidence, thus we don't "forget" it here.